java/sage/VideoFrame.java

```diff
@@ -5650,9 +5651,42 @@ private int getMatchingLangIndex(String[] langOptions, String targetLang)
 
   private boolean selectDefaultSubpicLanguage()
   {
+    String defaultAudioLang = uiMgr.get("default_audio_language", "English"); 
+    MediaLangInfo mli = (MediaLangInfo) mediaLangMap.get(defaultAudioLang);	
+    
     String defaultLang = uiMgr.get("default_subpic_language", "");
-    if (defaultLang == null || defaultLang.length() == 0)
-      return true; // preferred no subtitles
+    
+    //Check for forced subtitles if there is not a defaul subtitle language set
+    if(defaultLang == null || defaultLang.length() == 0)
+    {
+      String [] subs = getDVDAvailableSubpictures();
```

Comment:
Make a copy of this array, otherwise you're overwriting a cache used in other areas potentially.